packages/vitest/src/runtime/setup.ts

```diff
@@ -15,16 +16,31 @@ export async function setupGlobalEnv(config: ResolvedConfig) {
     enumerable: false,
   })
 
-  // it's useful to see the full stack trace in the console by default
-  Error.stackTraceLimit = 100
-
   // should be re-declared for each test
   // if run with "threads: false"
   setupDefines(config.defines)
 
   if (globalSetup)
     return
 
+  const state = getWorkerState()
+
+  installViteNodeSourcemaps({
+    getSourceMap(source) {
+      const fsPath = state.moduleCache.normalizePath(source)
+      const cache = state.moduleCache.get(fsPath)
+      if (cache.map)
+        return cache.map
+      const mapString = cache?.code?.match(/\/\/# sourceMappingURL=data:application\/json;charset=utf-8;base64,(.+)/)?.[1]
```

Comment:
Fun fact - when I used `sourcemap: true` and stored it inside `state.moduleCache` in `vite-node/client`, tests became 3x slower 😄 (I guess because `JSON.stringify/parse` between worker and main thread is very expensive on large maps 😄)